eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -2292,4 +2292,42 @@ default void appendString(Appendable appendable, String separator)
      * @since 3.0
      */
     <K, V> MapIterable<K, V> aggregateBy(Function<? super T, ? extends K> groupBy, Function0<? extends V> zeroValueFactory, Function2<? super V, ? super T, ? extends V> nonMutatingAggregator);
+
+    /**
+     * Applies a groupBy function over the iterable, followed by a collect function.
+     * The groupBy function creates a {@link Multimap} where the key is the transformed result
+     * and the actual element is the value. The collect function transforms the values in {@link Multimap}
+     * created by the groupBy function.
+     *
+     * @param groupByFunction a {@link Function} to use as the groupBy transformation function
+     * @param collectFunction a {@link Function} to use as the collect transformation function
+     *
+     * @return A {@link Multimap} where the key is the transformed result from applying the groupBy function
+     * and the value is the transformed result from applying the collect function.
+     *
+     * @see #groupBy(Function)
+     * @see Multimap#collectValues(Function)
+     *
+     * @since 10.1.0
+     */
+    default <K, V> Multimap<K, V> groupByAndCollect(Function<? super T, ? extends K> groupByFunction, Function<? super T, ? extends V> collectFunction)
+    {
+        Multimap<K, T> groupedMap = this.groupBy(groupByFunction);
+        return groupedMap.collectValues(collectFunction);
+    }
+
+    /**
+     * Same as {@link #groupByAndCollect(Function, Function)}, except that the results are gathered into the specified
+     * {@code target} multimap.
+     *
+     * @since 10.1.0
+     */
+    default <K, V, R extends MutableMultimap<K, V>> R groupByAndCollect(
+            Function<? super T, ? extends K> groupByFunction,
+            Function<? super T, ? extends V> collectFunction,
+            R target)
+    {
+        this.groupBy(groupByFunction).collectValues(collectFunction).forEachKeyMultiValues(target::putAll);
```

Comment:
It would be more efficient to use collectValues() that takes a target.

```java
<V2, R extends MutableMultimap<K, V2>> R collectValues(Function<? super V, ? extends V2> function, R target);
```

And even better would be a direct implementation like toMap()

```java
this.forEach(each -> target.put(keyFunction.apply(each), valueFunction.apply(each)));
return target;
```

Reply:
I'll remove the implementation that doesn't take a target multimap for now.

Do you mind if I update the implementation that takes a target multimap to the one in your comment?